lib/queuePopulator/KafkaLogConsumer/ListRecordStream.js

```diff
@@ -0,0 +1,90 @@
+const stream = require('stream');
+
+class ListRecordStream extends stream.Transform {
+    /**
+     * @constructor
+     * @param {Logger} logger logger
+     */
+    constructor(logger) {
+        super({ objectMode: true });
+        this._logger = logger;
+    }
+
+    /**
+     * Maps change stream operation type to
+     * backbeat supported types
+     * @param {string} operationType mongo opetation type
+     * @returns {string|undefined} supported operation type
+     */
+    _getType(operationType) {
+        switch (operationType) {
+            case 'insert':
+            case 'update':
+            case 'replace':
+                return 'put';
+            case 'delete':
+                return 'delete';
+            default:
+                this._logger.warn('Got unsupported operation', {
+                    method: 'ListRecordStream._getType',
+                    operationType,
+                });
+                return undefined;
+        }
+    }
+
+    /**
+     * Parse kafka message's
+     * value field twice, as it can be over stringified
+     * when reading messages pushed by Kafka-connect
+     * MongoDB source connector
+     * @param {string} blob sringified json
+     * @returns {Object} parsed json
+     */
+    _parseKafkaMessageValue(blob) {
+        try {
+            let parsed = JSON.parse(blob);
+            if (typeof parsed === 'string') {
+                parsed = JSON.parse(parsed);
+            }
+            return parsed;
```

Comment:
```suggestion
            return JSON.parse(JSON.parse(blob));
```
Just do a double parse immediately, we are parsing from a defined rule so the string check is redundant